service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java

```diff
@@ -686,6 +696,44 @@ public void verifyConnectionDurationLimitIsReached(final VertxTestContext ctx) {
                 }));
     }
 
+    /**
+     * Verifies that the (default) ConnectionEvent API configured for a protocol adapter
+     * sets the connection event message's TTL header value before forwarding the message
+     * to downstream aoplications.
+     * 
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    public void tesForwardedConnectionEventMessageHasTtlHeaderSet(final VertxTestContext ctx) {
+
+        // GIVEN a protocol adapter configured to send connection events
+        final ConnectionEventProducer connectionEventProducer = new HonoEventConnectionEventProducer();
+        adapter.setConnectionEventProducer(connectionEventProducer);
+        final DownstreamSender connectionEventSender = mock(DownstreamSender.class);
+        when(connectionEventSender.send(any(Message.class))).thenReturn(Future.succeededFuture());
+        when(downstreamSenderFactory.getOrCreateEventSender(Constants.DEFAULT_TENANT)).thenReturn(Future.succeededFuture(connectionEventSender));
+
+        // WHEN a device, belonging to a tenant for which a max TTL is configured, connects to such an adapter
+        final Device authenticatedDevice = new Device(Constants.DEFAULT_TENANT, "4711");
+        final TenantClient tenantClient = mock(TenantClient.class);
+        when(tenantClientFactory.getOrCreateTenantClient()).thenReturn(Future.succeededFuture(tenantClient));
+        final TenantObject tenantObject = TenantObject.from(Constants.DEFAULT_TENANT, true);
+        final ResourceLimits tenantLimits = mock(ResourceLimits.class);
```

Comment:
there is no need to mock this class, is there?

Reply:
no, not really. will be fixed